numpy/core/__init__.py

```diff
@@ -53,6 +53,23 @@
 del os
 
 from . import umath
+
+# Test that multiarray is a pure python module wrapping _multiarray_umath,
+# and not the old c-extension module.
+if not (hasattr(multiarray, '_multiarray_umath') and
+        hasattr(umath, '_multiarray_umath')):
+    from os.path import dirname
+    path1 = dirname(multiarray.__file)
+    path2 = dirname(umath.__file)
+    msg = ("Something is wrong with the NumPy installation. "
+        "While importing we detected an older version of "
+        "numpy in {}. One method of fixing this is to repeatedly uninstall "
+        "numpy until none is found, then reinstall this version.")
+    if path1 == path2:
+        raise ImportError(msg.format(path1))
+    else:
+        raise ImportError(msg.format('{} or {}'.format(path1, path2))
```

Comment:
I think showing specifically `np.__path__` would be useful here, as that will list all numpy installations.

Reply:
Good detective work - I knew the interpreter would have that info somewhere, but wasn't sure where.